taipy/core/_entity/submittable.py

```diff
@@ -86,7 +88,13 @@ def is_ready_to_run(self) -> bool:
         Returns:
             True if the given entity is ready to be run. False otherwise.
         """
-        return all(dn.is_ready_for_reading for dn in self.get_inputs())
+        if self._submittable_id not in SubmittableStatusCache.submittable_id_datanodes:
+            for dn in self.get_inputs():
+                if not dn.is_ready_for_reading:
+                    SubmittableStatusCache.add(self._submittable_id, dn.id)
+        if len(SubmittableStatusCache.submittable_id_datanodes[self._submittable_id]) == 0:
+            return True
+        return False
```

Comment:
What about delegating more by targeting the following code:

```suggestion
        if self._submittable_id not in SubmittableStatusCache:
            SubmittableStatusCache.add(self._submittable_id, self.get_inputs())
        return SubmittableStatusCache.is_ready_to_run(self._submittable_id)
```

I did another suggestion in the `SubmittableStatusCache` class to match this usage.

Reply:
I don't think it's good considering we're adding all input DNs without checking?

Reply:
I'm proposing a new design, will let you know once pushed